soccer/src/soccer/strategy/agent/position/defense.cpp

```diff
@@ -5,69 +5,112 @@ namespace strategy {
+        // attempt to pass the ball to the target robot
+        rj_geometry::Point target_robot_pos =
+            world_state()->get_robot(true, target_robot_id).pose.position();
+        planning::LinearMotionInstant target{target_robot_pos};
+        auto line_kick_cmd = planning::MotionCommand{"line_kick", target};
+        intent.motion_command = line_kick_cmd;
+        intent.shoot_mode = RobotIntent::ShootMode::KICK;
+        // NOTE: Check we can actually use break beams
+        intent.trigger_mode = RobotIntent::TriggerMode::ON_BREAK_BEAM;
+        // TODO: Adjust the kick speed based on distance
+        intent.kick_speed = 4.0;
+        intent.is_active = true;
+        return intent;
     }
+
+    return std::nullopt;
 }
 
-communication::PosAgentResponseWrapper Defense::receive_communication_request(
-    communication::AgentPosRequestWrapper request) {
-    communication::PosAgentResponseWrapper comm_response;
-    if (const communication::PassRequest* pass_request =
-            std::get_if<communication::PassRequest>(&request.request)) {
-        // TODO (https://app.clickup.com/t/8677c0q36): Handle pass response
-        sleep(5);
-        communication::Acknowledge acknowledge{};
-        communication::generate_uid(acknowledge);
-        comm_response.response = acknowledge;
-    } else if (const communication::PositionRequest* position_request =
-                   std::get_if<communication::PositionRequest>(&request.request)) {
-        communication::PositionResponse position_response{};
-        position_response.position = position_name_;
-        communication::generate_uid(position_response);
-        comm_response.response = position_response;
-    } else if (const communication::TestRequest* test_request =
-                   std::get_if<communication::TestRequest>(&request.request)) {
-        communication::TestResponse test_response{};
-        test_response.message = fmt::format("robot {} says hello", robot_id_);
-        communication::generate_uid(test_response);
-        comm_response.response = test_response;
-    } else {
-        communication::Acknowledge acknowledge{};
-        communication::generate_uid(acknowledge);
-        comm_response.response = acknowledge;
-    }
+communication::Acknowledge Defense::acknowledge_pass(
+    communication::IncomingPassRequest incoming_pass_request) {
+    // Call to super
+    communication::Acknowledge acknowledge_response =
+        Position::acknowledge_pass(incoming_pass_request);
+    // Update current state
+    current_state_ = FACING;
+    // Return acknowledge response
+    return acknowledge_response;
+}
+
+void Defense::pass_ball(int robot_id) {
+    // Call to super
+    Position::pass_ball(robot_id);
+    // Update current state
+    current_state_ = PASSING;
+}
 
-    return comm_response;
+communication::Acknowledge Defense::acknowledge_ball_in_transit(
+    communication::BallInTransitRequest ball_in_transit_request) {
+    // Call to super
```

Comment:
this is kind of a weak comment, I would prefer no comment over this, or an explanation of what to expect from acknowledge_response